### PR TITLE
fix jbig2dec download path

### LINKED ISSUE #11680: wolfi-bot/jbig2dec
failed to bump package jbig2dec to version 0.20: got 404 Not Found when fetching https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs9531/jbig2dec-0.20.tar.gz

### PR DESCRIPTION
**Pull request by found-it**
Fixes https://github.com/wolfi-dev/os/issues/11680

Looks like the download path changed.

### CODE REVIEW

File: jbig2dec.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ environment:
 pipeline:
   - uses: fetch
     with:
-      expected-sha256: 279476695b38f04939aa59d041be56f6bade3422003a406a85e9792c27118a37
-      uri: https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs9531/jbig2dec-${{package.version}}.tar.gz
+      expected-sha256: 7b63ff6470289547e7a3a0f145cb8ea6c2afffdd65645b7d87d3b7febc96fb3a
```
Comment (mamccorm) [inline jbig2dec.yaml:24]: [Non-blocking] But unless there is a reason for not using Git checkout, we should probably consider switching over. It looks like [the git repo cuts release tags](https://github.com/ArtifexSoftware/jbig2dec/tags), and it'd mean no issues with changing download URLs

**Review by mamccorm — DISMISSED** [review]
Added one comment but non-blocking

**Comment by found-it** [comment]
Good call out @mamccorm, I'll switch it to the tag. The repo description seems to call out the tags page in a confusing way but I think that's the best move

> This is a mirror: the canonical repo is: git.ghostscript.com/jbig2dec.git. This repo does not host releases, they are here: https://github.com/ArtifexSoftware/jbig2dec/tags

File: jbig2dec.yaml [hunk 2]
```diff
@@ -19,10 +19,11 @@ environment:
...
+  - uses: git-checkout
     with:
-      expected-sha256: 279476695b38f04939aa59d041be56f6bade3422003a406a85e9792c27118a37
-      uri: https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs9531/jbig2dec-${{package.version}}.tar.gz
+      repository: https://github.com/ArtifexSoftware/jbig2dec.git
+      expected-commit: 1d1347e38a55e657dcc4c8f1c77bb3a26bfc9ff3
+      tag: ${{package.version}}
 
   - runs: |
       autoreconf -vif
```
Comment (ajayk) [inline jbig2dec.yaml:29]:
```suggestion
      ./autogen.sh
      autoreconf -vif
```

Comment (found-it) [reply jbig2dec.yaml:29]: beat me to it :)

**Review by ajayk — APPROVED** [review]

**Pull request merged**